tests/spicy/types/unit/unit-max-size-as-field.spicy

```diff
@@ -0,0 +1,47 @@
+# @TEST-DOC: Tests &max-size on units when not fully consumed from a field.
+#
+# @TEST-EXEC: printf '123' | spicy-driver -d %INPUT >output
+# @TEST-EXEC: btest-diff output
+
+module Test;
+
+public type Outer = unit {
+    dat: Inner;
+    b  : uint8[] &size=2;
```

Comment:
Nit: IMO the test baselines become more readable if we parsed `bytes &size=1` instead of `uint8`. Here and below.